refactor: webpack config

devCrossNet opened the pull request:
<!--
There are two main goals in this document, depending on the nature of your PR:

- description: please tell us about your PR
- checklist: please review the checklist

To help to quickly understand the nature of your pull request,
please create a description that incorporates the following elements:
-->

### What is accomplished by your PR?
This PR hopefully simplifies the way we manage the webpack configs.

### Is there something controversial in your PR?
not that I know


### Link to the Issue
no ticket

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [ ] Add new passing unit tests to cover the code introduced by your PR
- [ ] Add new documentation for the code introduced by your PR


<!--
Thanks for contributing!
-->

tamer-mohamed commented on docs/guide/third-party.md:
```diff
@@ -3,11 +3,13 @@
 ### Configure Webpack
 
-`./config/webpack.base.config.js`:
+`./config/base.js`:
```
how about renaming `config` folder to `webpack`? more explicit.

devCrossNet replied:
I like it 👍

tamer-mohamed commented on config/base-server.js:
```diff
@@ -0,0 +1,30 @@
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      CLIENT: false,
```
We can use one or the other, `CLIENT: true | false` and get rid of `SERVER`.
`CLIENT` is enough to determine the context whether it is client or server, IMO.

devCrossNet commented:
@tamer-mohamed I moved the whole webpack thing to the tools folder (we could potentially break this out into its own package later). I also changed the webpack-configs to typescript. 

Can you do me the favour to test this PR before I merge it? (dev, build, test, etc.)

tamer-mohamed commented:
@devCrossNet `build`  `dev` and `test` tasks works as it should.

About the packages, I have a proposal for the packages and working solution, we should discuss it soon.
